src/main/java/com/google/googleidentity/filter/UserAuthenticationFilter.java

```diff
@@ -0,0 +1,47 @@
+package com.google.googleidentity.filter;
+
+
+
+import com.google.googleidentity.security.UserSession;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+
+
+import javax.servlet.*;
+import javax.servlet.http.*;
+import java.io.IOException;
+
+@Singleton
```

Comment:
Add javadoc comments to public classes to summarize what they do, here and elsewhere.

https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required